### PR TITLE
Updates for MPM version 1.3.1.78

### PR DESCRIPTION
**Pull request by MRC3742**
Updates to bring Companion and Radio GUI up to current MPM release 1.3.1.78 and work in progress for commits up to 1.3.1.80
Not sure if Radio GUI updates are needed anymore as it is being sent from the MPM?

@3djc please review

Thanks - Rich

### CODE REVIEW

**Comment by 3djc** [comment]
Hello @MRC3742 , yes, I think it is still worth doing; it allows you to setup an external module, or look at what protocol you selected, while the module is not inserted. Probably not critical, but nice to have !. Will review it today

**Comment by MRC3742** [comment]
> 
> 
> Hello @MRC3742 , yes, I think it is still worth doing; it allows you to setup an external module, or look at what protocol you selected, while the module is not inserted. Probably not critical, but nice to have !. Will review it today

@3djc, After testing I see a possible issue with Radio GUI.
Testing was done on a T16, external module selected but no module inserted.
When a Protocol is selected that has only one subtype (and is listed correctly in multiprotocols.cpp subtype table) the subtype is left blank in the Radio GUI. Examples are Futaba, FX816, Propel, etc.
This works properly with MPM inserted or using the internal MPM.

It did work as expected on earlier OTX versions but have not tested since the information has been sent to OTX via the MPM. If you feel this needs repair, I can open an issue and work to find the commit that caused it to go blank.

Thanks - Rich

**Comment by 3djc** [comment]
I'll fix it @MRC3742

**Comment by 3djc** [comment]
@MRC3742 Fixed

File: companion/src/firmwares/moduledata.h [hunk 1]
```diff
@@ -135,7 +135,14 @@ enum MultiModuleRFProtocols {
   MODULE_SUBTYPE_MULTI_DSM_RX,
   MODULE_SUBTYPE_MULTI_JJRC345,
   MODULE_SUBTYPE_MULTI_Q90C,
-  MODULE_SUBTYPE_MULTI_LAST = MODULE_SUBTYPE_MULTI_Q90C
+  MODULE_SUBTYPE_MULTI_KYOSHO,
+  MODULE_SUBTYPE_MULTI_RLINK,
+  MODULE_SUBTYPE_MULTI_ELRS,
```
Comment (rabbitambulance-ra) [inline companion/src/firmwares/moduledata.h:140]: @MRC3742, sorry for the necromancy, but could you please share what the idea for `MODULE_SUBTYPE_MULTI_ELRS` was? Surely this isn't ExpressLRS, right?

Comment (MRC3742) [reply companion/src/firmwares/moduledata.h:140]: Yes, ELRS was assigned a protocol slot in multi software and was going to be implemented with an additional add on rfm95w module.
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/pull/404

This PR was eventually CLOSED and never implemented as ELRS prebuilt modules became widely available.

Rich

**Pull request merged**